event-buffer/event-buffer-core/src/main/java/uk/gov/justice/services/event/buffer/core/repository/service/ConsecutiveEventBufferService.java

```diff
@@ -0,0 +1,135 @@
+     * according to the stream_status repository then an empty stream is returned and the incomingEvent
+     * is added to the event-buffer-repository. If the incomingEvent is consecutive then it is returned
+     * as a stream with any consecutive events from the buffer. If an event is the first to be processed
+     * for that streamId then the version value must be 1 or the incomingEvent is added to the buffer
+     * and an empty stream is returned.
+     *
+     * @param incomingEvent
+     * @return streamOfConsecutiveEvents
+     */
+    @Override
+    public Stream<JsonEnvelope> currentOrderedEventsWith(final JsonEnvelope incomingEvent) {
+
+        LOGGER.trace(format("Message buffering for message: %s", toEnvelopeTraceString(incomingEvent)));
+
+        final UUID streamId = incomingEvent.metadata().streamId().get();
+        final Long incomingEventVersion = incomingEvent.metadata().version().get();
+
+        final Optional<StreamStatus> currentStatus = streamStatusRepository.findByStreamId(streamId);
+        if (currentStatus.isPresent()) {
+            final Long currentVersion = currentStatus.get().getVersion();
+            if (incomingEventObsolete(incomingEventVersion, currentVersion)) {
+                LOGGER.trace(format("Message : %s is an obsolete version", toEnvelopeTraceString(incomingEvent)));
+                return Stream.empty();
+
+            } else if (incomingEventNotInOrder(incomingEventVersion, currentVersion)) {
+                LOGGER.trace(format("Message : %s is not consecutive, adding to buffer", toEnvelopeTraceString(incomingEvent)));
+                return addToBufferAndReturnEmptyStream(incomingEvent, streamId, incomingEventVersion);
+
+            } else {
+                LOGGER.trace(format("Message : %s version is valid sending stream to dispatcher", toEnvelopeTraceString(incomingEvent)));
+                return streamOfIncomingAndBufferedEvents(incomingEvent, streamId, incomingEventVersion);
+
+            }
+        } else {
+            if(incomingEventVersion == 1) {
+                LOGGER.trace(format("Message : %s is a new streamId registering with stream_status repo and sending to dispatcher", toEnvelopeTraceString(incomingEvent)));
+                streamStatusRepository.insert(new StreamStatus(streamId, incomingEventVersion));
+                return Stream.of(incomingEvent);
+            } else {
+                LOGGER.trace(format("Message : %s is a new streamId but version is not one so adding to buffer", toEnvelopeTraceString(incomingEvent)));
+                return addToBufferAndReturnEmptyStream(incomingEvent, streamId, incomingEventVersion);
+            }
+        }
+
+    }
+
+    private Stream<JsonEnvelope> streamOfIncomingAndBufferedEvents(JsonEnvelope incomingEvent, UUID streamId, Long incomingEventVersion) {
+        streamStatusRepository.update(new StreamStatus(streamId, incomingEventVersion));
```

Comment:
If this is a new StreamStatus (with an existing streamId) won't this be a detached entity, will the update resolve properly? Will it merge with the existing matching UUID or fail at the repo level?